src/Client/MultiplexedConnections.cpp

```diff
@@ -131,16 +131,29 @@ void MultiplexedConnections::sendQuery(
             modified_settings.group_by_two_level_threshold = 0;
             modified_settings.group_by_two_level_threshold_bytes = 0;
         }
+
+        if (settings.parallel_reading_from_replicas)
+        {
+            assert(replica_info != ReplicaInfo{});
+            modified_settings.collaborate_with_initiator = true;
+            modified_settings.count_participating_replicas = replica_info.all_replicas_count;
+            modified_settings.number_of_current_replica = replica_info.number_of_current_replica;
```

Comment:
I don't like we add so many settings, because they are visible to user.
Maybe we can send it at connection start? E.g. in client info.